src/syscall_hook.h

```diff
@@ -0,0 +1,45 @@
+#ifndef CORE_SYSCALL_HOOK_H
+#define CORE_SYSCALL_HOOK_H
+
+#define _GNU_SOURCE
+#include <dlfcn.h>
```

Comment:
Reduce the header inclusion. Don't expose too many macros/prototypes in advance.